### PR TITLE
batou migrate

### PR DESCRIPTION
**Pull request by icemac**
Partial fix for #185.

Open issues:

- [x] ~~add migration needed for #185~~ done in #255
- [x] add documentation for other manual steps
- [x] force migration if outdated
- [x] create configuration file when bootstrapping batou

### CODE REVIEW

File: src/batou/migrate/migrations/2300.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def migrate(output):
+    """Manual upgrade steps for 2.3."""
+    output(
+        'Change Address init parameters', """
```
Comment (ctheune) [inline src/batou/migrate/migrations/2300.py:4]: i think there's a dedent missing here.

File: src/batou/migrate/migrations/2300.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        The new default DNS resolve scheme will resolve IPv4 by default. If
+        you want IPv6 you have to set that explicitly. Please adapt all
+        occurrences of `Address` in you components and explicitly set
+        `require_v6` and `require_v4`.""")
+    output(
+        'Colliding attributes in environment and secrets',
+        'If an attribute exists in secrets and also in plain text'
+        ' environment config, batou fails. Delete it from plain text.')
+    output(
+        '`Attribute` requires explicit default.', """
```
Comment (ctheune) [inline src/batou/migrate/migrations/2300.py:14]: i think there's a dedent missing here.

File: src/batou/migrate/migrations/2300.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    """Manual upgrade steps for 2.3."""
+    output(
+        'Change Address init parameters', """
+        The new default DNS resolve scheme will resolve IPv4 by default. If
+        you want IPv6 you have to set that explicitly. Please adapt all
+        occurrences of `Address` in you components and explicitly set
+        `require_v6` and `require_v4`.""")
+    output(
+        'Colliding attributes in environment and secrets',
+        'If an attribute exists in secrets and also in plain text'
```
Comment (ctheune) [inline src/batou/migrate/migrations/2300.py:11]: maybe make this a dedented multi-line string too, for consistency.

**Review by ctheune — CHANGES_REQUESTED** [review]
Looks really good, thanks! There's a minor formatting issue that I'd like to get checked in the 2300 migration, otherwise' I'd be happy to merge this.

File: src/batou/migrate/__init__.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+from batou._output import TerminalBackend, output
+
+CONFIG_FILE_NAME = '.batou.json'
+MIGRATION_MODULE = 'batou.migrate'
+
+
+def output_migration_step(title: str, text: str) -> None:
+    """Print the information of migration step in a formatted way."""
+    output.section(title, red=True)
+    output.line(textwrap.dedent(text).replace('\n', ''))
```
Comment (icemac) [inline src/batou/migrate/__init__.py:17]: This line does the dedent for the migration steps.

File: src/batou/migrate/migrations/2300.py (hunk 1 above)
Comment (icemac) [reply src/batou/migrate/migrations/2300.py:4]: No, it is done in `output_migration_step()`

**Comment by ctheune** [comment]
Hrgh. Looks like we need a rebase.

**Comment by icemac** [comment]
Merged `master` into this branch, resolving the conflicts.

**Pull request merged**